In [21]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [22]:
Brady_Bunch = pd.DataFrame()

In [23]:
Brady_Bunch["age"] = [14, 12, 11, 10, 8, 7, 8]
Brady_Bunch["name"] = ["Greg", "Marcia", "Peter", "Jan", "Bobby", "Cindy", "Oliver"]

In [24]:
import statistics

statistics.mode(Brady_Bunch["age"])

8

In [25]:
Brady_Bunch.describe()

age
count   7.000000
mean   10.000000
std     2.516611
min     7.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [26]:
import statistics

np.median(Brady_Bunch["age"])



10.0

In [27]:
np.var(Brady_Bunch.age)

5.428571428571429

In [28]:
np.std(Brady_Bunch["age"], ddof=1) / np.sqrt(len(Brady_Bunch["age"]))

0.9511897312113418

If I had to choose just one measure of central tendency to describe the data, I would choose the median (10). In this case, the age in the middle of the sorted list is most representative of the data.

If I had to choose just one estimate of variance to describe the data, I would choose the standard error (1.033). This means that the mean that we calculated for our data is only off by a little over one year.

After Cindy has a birthday, the median remains the same and so does the mode. The mean and the estimates of variance shift slightly. The standard error got even smaller (under one year).

In [29]:
BB = pd.DataFrame()
BB["age"] = [14, 12, 11, 10, 8, 7, 1]
BB["name"] = ["Greg", "Marcia", "Peter", "Jan", "Bobby", "Cindy", "Jessica"]

In [30]:
statistics.mode(BB["age"])

StatisticsError: no unique mode; found 7 equally common values

In [31]:
BB.describe()

age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000

In [32]:
np.std(BB["age"], ddof=1) / np.sqrt(len(BB["age"]))

1.6035674514745462

With the replacement of Oliver (age 8) with Jessica (age 1), there is no longer a mode. The median is still 10, but the mean (age 9) is the measure of central tendency that I would use to describe this new data set. As for the estimate of variance, I would still use the standard error (even though it increased by about half a year).

In [33]:
fans = pd.DataFrame()

fans["mags"] = ["TV Guide", "Entertainment Weekly", "Pop Culture Today", "SciPhi Fanatic"]
fans["percentage"] = [.2, .23, .17, .05]

fans

mags  percentage
0              TV Guide        0.20
1  Entertainment Weekly        0.23
2     Pop Culture Today        0.17
3        SciPhi Fanatic        0.05

In [34]:
fans.describe()

percentage
count    4.000000
mean     0.162500
std      0.078899
min      0.050000
25%      0.140000
50%      0.185000
75%      0.207500
max      0.230000

In [36]:
fans.median()

percentage    0.185
dtype: float64

Based on this data on the 50th anniversary, I would estimate that 18% of Americans were Brady Bunch fans. On the one hand, people who subscribe to SciPhi Fanatic are less likely to watch the Brady Bunch. At the same time, this subpopulation should skew the data to a degree because not everyone enjoys pop culture.